vyper/semantics/analysis/module.py

```diff
@@ -105,8 +106,6 @@ def __init__(
         # internal functions are intentionally included in this check, to prevent breaking
         # changes in in case of a future change to their calling convention
```

Comment:
The current state of the PR removes the check for internal functions signatures collision, in such case, the comments are not needed anymore
```suggestion
```